generate_commitment/generate_submarine_commit.py

```diff
@@ -161,35 +179,149 @@ def generateAddressBexport(fromAddress, toAddress, sendAmount, dappData, gasPric
+    parser.add_argument(
+        '-l',
+        '--gas-limit',
+        required=False,
+        type=int,
+        default=3712394,
+        help=
+        "Optional Gas limit for TX Unlock transaction. Default is 3.7 MWei.")
+    return parser.parse_args()
 
 
 def main():
+    '''
+    Main method. Runs the program if it is used standalone (rather than as an exported library).
+    '''
 
-    #init test variables
-    gasPrice = 50000000000  # 50 gwei
-    gasLimit = 3712394  # startgas
-    toAddress = normalize_address("0x7AEB1Fd3A42731c4Ae80870044C992eb689fb2Fe")
-    fromAddress = normalize_address("0x94146296881b3322838cFF5f2d50fdde841928D8")
-    sendAmount = 123000000000000
-    dappData = b""
+    parser = _get_args()
+
+    if len(parser.target_address) != 42:
+        log.error(
+            "Target Address length does not appear to match the correct length of an Ethereum address"
+        )
+        sys.exit(1)
+    if len(parser.from_address) != 42:
+        log.error(
+            "From Address length does not appear to match the correct length of an Ethereum address"
+        )
+        sys.exit(2)
+    if parser.target_address[0:2] != "0x":
+        log.error(
+            "Target address not in expected format, expected address to start with 0x"
+        )
+        sys.exit(3)
+    if parser.from_address[0:2] != "0x":
+        log.error(
+            "From address not in expected format, expected address to start with 0x"
+        )
+        sys.exit(4)
+    hex_alphabet = [
+        "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d",
+        "e", "f"
+    ]
+    for hexChar in parser.target_address[2:].lower():
+        if hexChar not in hex_alphabet:
+            log.error(
+                "Found character in target address that does not match hex value: {}".
+                format(hexChar))
+            sys.exit(5)
+    for hexChar in parser.from_address[2:].lower():
+        if hexChar not in hex_alphabet:
+            log.error(
+                "Found character in from address that does not match hex value: {}".
+                format(hexChar))
+            sys.exit(6)
+
+    gasPrice = parser.gas_price
+    gasLimit = parser.gas_limit
+    toAddress = normalize_address(parser.target_address)
+    fromAddress = normalize_address(parser.from_address)
+    sendAmount = parser.amount
+    if (parser.dapp_data):
+        dappData = rec_bin(parser.dapp_data)
+    else:
+        dappData = b""
 
-    tx, addressB, commit, randw = generateAddressB(fromAddress, toAddress, sendAmount, dappData, gasPrice, gasLimit)
+    tx, addressB, commit, randw = _generateAddressBInternal(
+        fromAddress, toAddress, sendAmount, dappData, gasPrice, gasLimit)
 
     # print("-"* 35)
     # printRemix(fromAddress, tx, encode_hex(randw))
-    print("-"* 35)
+    print("-" * 35)
 
-    print("AddressB: %s" % addressB)  # addressB also can retreived using tx.to_dict().get("sender")
+    print("AddressB: %s" % addressB
```

Comment:
Can we standardize on the new `"{}".format(foo)` style?

Reply:
addressed